mevshare/builders.go

```diff
@@ -0,0 +1,238 @@
+package mevshare
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ybbus/jsonrpc/v3"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+type BuilderAPI uint8
+
+const (
+	BuilderAPIRefRecipient BuilderAPI = iota
+	BuilderAPIMevShareBeta1
+)
+
+var ErrInvalidBuilder = errors.New("invalid builder specification")
+
+type BuildersConfig struct {
+	Builders []struct {
+		Name     string `yaml:"name"`
+		URL      string `yaml:"url"`
+		API      string `yaml:"api"`
+		Internal bool   `yaml:"internal"`
```

Comment:
nitpick (non-blocking): Might be worth adding `omitempty` for attributes that default to false to simplify configuration, e.g

```suggestion
		Internal bool   `yaml:"internal,omitempty"`
```

Reply:
Added. But I think its more relevant for serialization.